### PR TITLE
Create stream_friendly_editing.js

### PR DESCRIPTION
**Pull request by TimeForANinja**
yet another example 😉

### CODE REVIEW

**Comment by fent** [comment]
ah nice! what do you think of replacing the other ffmpeg.js example with this one completely?

there are also a couple of lint errors

**Comment by TimeForANinja** [comment]
> ah nice! what do you think of replacing the other ffmpeg.js example with this one completely?
> 
> there are also a couple of lint errors

we'd miss out on the progress tracking (at least atm) and i haven't done much field testing but fine with me 🤷 
i prefer not having to dump everything into files on disk

**Comment by fent** [comment]
oh yeah progress tracking. would be cool to have multi-stream progress tracking. doesn't have to be super fancy progress bar, just a percentage, that something you want to add?



> i prefer not having to dump everything into files on disk

that was my intention with the ffmpeg.js example, but didn't know how

**Comment by TimeForANinja** [comment]
there we gone
* replacing old ffmpeg example instead of just adding a new file
* added progress tracker
* commented a bit more

stumbled about the possibility of working with streams when trying to fix a bug in fluent-ffmpeg
still don't understand why they only allow for a single input stream

File: example/ffmpeg.js [hunk 1]
```diff
@@ -1,47 +1,80 @@
...
-      .save(mainOutput)
-      .on('error', console.error)
-      .on('end', () => {
-        fs.unlink(audioOutput, err => {
-          if (err) console.error(err);
-          else console.log(`\nfinished downloading, saved to ${mainOutput}`);
-        });
-      });
-  });
+  process.stdout.write(`Audio | ${(tracker.audio.downloaded / tracker.audio.total * 100).toFixed(2)}% processed`);
```
Comment (fent) [inline example/ffmpeg.js:37]: missing a space at the end

File: package-lock.json [hunk 2]
```diff
@@ -1724,9 +1724,9 @@
       }
     },
     "lodash": {
-      "version": "4.17.15",
-      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.15.tgz",
-      "integrity": "sha512-8xOcRHvCjnocdS5cpwXQXVzmmh5e5+saE2QGoeQmbKmRS6J3VQppPOIt0MnmE+4xlZoumy0GPG0D0MVIQbNA1A==",
+      "version": "4.17.19",
+      "resolved": "https://registry.npmjs.org/lodash/-/lodash-4.17.19.tgz",
+      "integrity": "sha512-JNvd8XER9GQX0v2qJgsaN/mzFCNA5BRe/j8JN9d+tWyGLSodKQHKFicdwNYzWwI3wjRnaKPsGj1XkBjx/F96DQ==",
```
Comment (fent) [inline package-lock.json:1729]: not needed for the PR

Comment (TimeForANinja) [reply package-lock.json:1729]: will revert my `npm audit fix`

**Comment by fent** [comment]
thanks!

**Pull request merged**